SW-1931 lazy load design store

khaledsherkawi opened the pull request:
SW-1931 Add healthcheck_url to the material store model
 
SW-1931 Add healthcheck_url to the settings service

SW-1931 Remove info logging
 
SW-1931 Add and modify unit tests
 
SW-1931 Modify material store HEAD call to use the healthcheck API
 
SW-1931 Fix a postMessage console error
 
SW-1931 Load design store once and not on every title item click

SW-1931 Initialize variable at the beginning of the method

SW-1931 Modify design store template

SW-1931 Add design store loading spinner styling

SW-1931 Fix bug

SW-1931 Move the design store template into the tabs folder

SW-1931 Lazy load the design store

SW-1931 Modify design store initialization logic
This change will show the network issues page if the device is offline. However, this will not change if the device gets online until the user refreshes the page.

Commit:
SW-1931 Add healthcheck_url to the material store model

Commit:
SW-1931 Add healthcheck_url to the settings service

Commit:
SW-1931 Remove info logging

Commit:
SW-1931 Add and modify unit tests

Commit:
SW-1931 Modify material store HEAD call to use the healthcheck API

Commit:
SW-1931 Fix a postMessage console error

Commit:
SW-1931 Load design store once and not on every title item click

Yelko-mrbeam commented on octoprint_mrbeam/static/js/material-store.js:
```diff
@@ -70,12 +71,15 @@ $(function () {
-                    data,
-                    self.material_store_iframe_src
-                );
+            let materialStoreIframeElement = $("#material_store_iframe");
```
if we want to have a variable we could then also initialise it at the beginning of the method. I usually don't like to initialise jquery elements because sometimes it leads to unexpected outcomes.. For example on success of an ajax call sometimes the variable could just be empty for reasons.

khaledsherkawi replied:
done

Yelko-mrbeam commented on octoprint_mrbeam/templates/mrbeam_ui_index.jinja2:
```diff
@@ -51,9 +51,9 @@
                     {# Translatots: Name of the tab #}
 					<li><a href="#designlib" data-toggle="tab" id="designlib_tab_btn">{{ _('design library') }}</a></li>
-                    <li><a href="#designstore" data-toggle="tab" id="designstore_tab_btn" onclick="window.mrbeam.viewModels.designStoreViewModel.goToStore()">{{ _('design store') }}</a></li>
+                    <li><a href="#designstore" data-toggle="tab" id="designstore_tab_btn" onclick="window.mrbeam.viewModels.materialStoreViewModel.initialiseStore()">{{ _('design store') }}</a></li>
```
This seems wrong.. we are calling initilaliseStore() from materialStoreViewModel... We are also not adding the lazy loading tag on the iframe.... Is this code review ready?

khaledsherkawi replied:
no it wasn't ready

Yelko-mrbeam reviewed (changes requested):
Material store seems almost finished. Only one extra detail to be added is the healthcheck_url on the repr() method. Design store code seems to be missing @khaledsherkawi ?

Commit:
SW-1931 Initialize variable at the beginning of the method

Commit:
SW-1931 Modify design store template

Commit:
SW-1931 Add design store loading spinner styling

Commit:
SW-1931 Fix bug

Commit:
SW-1931 Move design store template into the tabs folder

Commit:
SW-1931 Lazy load the design store

Commit:
SW-1931 Modify design store initialization logic

This change will show the network issues page if the device is offline. However, this will not change if the device gets online until the user refreshes the page.

Yelko-mrbeam commented on octoprint_mrbeam/static/js/design_store.js:
```diff
@@ -279,8 +296,8 @@ $(function () {
             }, 3000);
-            document.getElementById("design_store_iframe").src =
-                self.DESIGN_STORE_IFRAME_SRC;
+            document.getElementById("design_store_iframe").src = "#";
```
why are we setting the src to # all the time on reload? 😕 is this working?

khaledsherkawi replied:
Yes because the page is set not to be reloaded every time we click on the "design store" item in the menu based on the element source. So if we didn't set it to "#" it will not initialise

khaledsherkawi replied:
@Yelko-mrbeam

Yelko-mrbeam commented on octoprint_mrbeam/static/js/design_store.js:
```diff
@@ -18,8 +19,21 @@ $(function () {
+                // This will show the network issue page if the device is offline
+                // However, if the device gets online afterwards, this will not change
+                // until the user refreshes the page
+                if(!window.mrbeam.isOnline) {
```
👍

Yelko-mrbeam commented on octoprint_mrbeam/static/js/material-store.js:
```diff
@@ -65,17 +66,19 @@ $(function () {
-                    data,
-                    self.material_store_iframe_src
-                );
+            if(materialStoreIframeElement.is(":visible")){
```
but if you are calling this method the iframe should be visible right? otherwise you shouldn't call this method... I think it's ok to break here if it's not visible, so that we catch the bug as soon as possible... at least an else  bracket console logging the error

khaledsherkawi replied:
resolved

khaledsherkawi commented:
> Material store seems almost finished. Only one extra detail to be added is the healthcheck_url on the repr() method. Design store code seems to be missing @khaledsherkawi ?

@Yelko-mrbeam I'm not sure what you mean by that. Is this comment still viable??

Commit:
SW-1931 Fix condition and add console error handling

The error is technically prompt by the window object of the Iframe, so better handle it based on that.
